web_monitoring/db.py

```diff
@@ -1,47 +1,657 @@
 # Functions for interacting with web-monitoring-db
+from dateutil.parser import parse as parse_timestamp
 import json
 import os
 import requests
+import requests.exceptions
+import time
+import toolz
+import tzlocal
+import warnings
 
 
-# mutable singleton for stashing web-monitoring-db url, potentially other stuff
-settings = {}
-# At import time, grab settings from env if possible. User can always override.
-settings['db_url'] = os.environ.get('WEB_MONITORING_DB_URL')
-settings['db_email'] = os.environ.get('WEB_MONITORING_DB_EMAIL')
-settings['db_password'] = os.environ.get('WEB_MONITORING_DB_PASSWORD')
+def _tzaware_isoformat(dt):
+    """Express a datetime object in timezone-aware ISO format."""
+    if dt.tzinfo is None:
+        # This is naive. Assume they mean this time in the local timezone.
+        dt = dt.replace(tzinfo=tzlocal.get_localzone())
+    return dt.isoformat()
 
-WEB_MONITORING_CREATE_IMPORT_API = '{db_url}/api/v0/imports'
-WEB_MONITORING_SHOW_IMPORT_API = '{db_url}/api/v0/imports/{import_id}'
 
+class WebMonitoringDbError(Exception):
+    ...
+
+
+def _process_errors(res):
+    # If the app gives us errors, raise a custom exception with those.
+    # If not, fall back on requests, which will raise an HTTPError.
+    if res.ok:
+        return
```

Comment:
This might be the way to go, but I don't feel 100% sure that we'll always be interested in JSON (we might, for example, be interested in proxied content), and the individual functions have more information about what they want/expect. Could refactor later if this doesn't pan out.

Reply:
Oh, that is a very good point. :thumbsup: